CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.21)
```

Comment:
I see the need to increase this, because of https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries 
Shouldn't 3.12 be enough?

I did not test it yet, but just from looking at the diff this looks great, thanks @bwrsandman

Reply:
Using `OBJECT` is 3.12 but [using `TARGET_OBJECTS` with them is 3.21](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#id9)

Reply:
It's clear now, thanks for the explanation.